packages/lib/universal/upload/put-file.ts

```diff
@@ -12,14 +15,36 @@ type File = {
   arrayBuffer: () => Promise<ArrayBuffer>;
 };
 
+/**
+ * Uploads a document file to the appropriate storage location and creates
+ * a document data record.
+ */
+export const putDocumentFile = async (file: File) => {
+  const isEncryptedDocumentsAllowed = await getFlag('app_allow_encrypted_documents');
+
+  // This will prevent uploading encrypted PDFs or anything that can't be opened.
+  if (!isEncryptedDocumentsAllowed) {
+    await PDFDocument.load(await file.arrayBuffer()).catch((e) => {
+      console.error(`PDF upload parse error: ${e.message}`);
+
+      throw new AppError('INVALID_DOCUMENT_FILE');
+    });
+  }
+
+  const { type, data } = await putFileInDatabase(file);
+
+  return await createDocumentData({ type, data });
```

Comment:
I'd instead have this `return await putFile(...)` and remove the other changes

I'd probably also rename this to `putPdfFile`

Reply:
Opps yeah `putFileInDatabase` was supposed to be `putFile`

I'm not sure what you mean by remove the rest though?